### DISCUSSION TITLE
How to execute obfuscated script for mitmproxy Python API

### DISCUSSION THREAD

**Discussion by spaul-github** [mitmproxy/mitmproxy#7090] [Q&A]
**Problem statement:**
mitmproxy is not intercepting with my obfuscated script 

**Detailed Description:**
I'm using mitmproxy as a interceptor and using this script
proxy.py
```
from mitmproxy import http, ctx
class CopilotProxy:
    async def request(self, flow):
         print("Inside request")
    async def response(self, flow):
          print("Inside response")
addons = [
Copilotproxy()
]
```
and i'm executing it with port 9001:
`mitmdump -p 9001 -s proxy.py`

but when  I obfuscated that proxy.py script using obfuscation tools like `PyInstaller` or `PyArmor`, no request is passing through the request function but it's working with out obfuscated.

**Comment by mhils** [comment]
That's a problem with your obfuscation, not a problem with mitmproxy.

**Reply by spaul-github** [reply]
Hi @mhils , obfuscation with python script is not a issue but I need to understand will mitmproxy work with a obfuscated script as I mentioned ?
